backend/fms_core/resources/sample.py

```diff
@@ -161,21 +161,31 @@ def import_obj(self, obj, data, dry_run):
 
         # TODO: This should throw a nicer warning if the individual already exists
         # TODO: Warn if the individual exists but pedigree/cohort is different
-        try:
-            individual, individual_created = Individual.objects.get_or_create(
-                name=get_normalized_str(data, "Individual ID"),
-                sex=get_normalized_str(data, "Sex", default=Individual.SEX_UNKNOWN),
-                taxon=taxon,
-                **({"pedigree": pedigree} if pedigree else {}),
-                **({"cohort": cohort} if cohort else {}),
-                **({"mother": mother} if mother else {}),
-                **({"father": father} if father else {}),
-            )
-            obj.individual = individual
-        except Exception as e:
-            individual_created = False
-            individual = None
-            errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
+
+        # If there's the individual's complete information then get/create
+        individual_is_complete = True if data["Individual ID"] and data["Sex"] and taxon else False
+        partial_individual = True if data["Individual ID"] or data["Sex"] or mother or father or taxon or pedigree or cohort else False
+        individual = None
+
+        if individual_is_complete:
```

Comment:
There is the case also where the individual already exists in which case the individual ID would be sufficient i think...
Something like
individual_already_exists = ID and not (all other individual fields)
if individual_already_exists or individual_is_complete :

Reply:
In that case it would be necessary to make sure the individual actually exists by checking in the DB and not just ensuring the ID is provided, right?

Reply:
Yeah if you want to provide a clear error message you would need to break down the get_or_create into a get and a create otherwise the error returned will be more generic (as it is atm).